packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -249,25 +256,40 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
      * @return Whether the bond has been successfully unbonded.
      */
     function _tryUnbond() private returns (bool) {
-        uint256 outputIndex = nextUnbondOutputIndex;
+        uint256 outputIndex;
+        uint128 bondAmount;
+        Types.Bond storage bond;
+        Types.CheckpointOutput memory output;
+
+        uint256 i = 0;
+        for (; i < MAX_UNBOND; ) {
+            outputIndex = nextUnbondOutputIndex + i;
+            bond = bonds[outputIndex];
+            bondAmount = bond.amount;
 
-        Types.Bond storage bond = bonds[outputIndex];
-        uint128 bondAmount = bond.amount;
-        if (block.timestamp >= bond.expiresAt && bondAmount > 0) {
-            delete bonds[outputIndex];
+            if (block.timestamp >= bond.expiresAt && bondAmount > 0) {
+                delete bonds[outputIndex];
 
-            Types.CheckpointOutput memory output = L2_ORACLE.getL2Output(outputIndex);
-            _increaseBalance(output.submitter, bondAmount);
-            emit Unbonded(outputIndex, output.submitter, bondAmount);
+                output = L2_ORACLE.getL2Output(outputIndex);
+                _increaseBalance(output.submitter, bondAmount);
+                emit Unbonded(outputIndex, output.submitter, bondAmount);
 
-            unchecked {
-                ++nextUnbondOutputIndex;
+                // Send reward message to L2 ValidatorRewardVault.
+                _sendRewardMessageToL2Vault(output);
+
+                unchecked {
+                    ++i;
+                }
+            } else {
+                break;
             }
+        }
+
+        if (i > 0 && output.timestamp > 0) {
```

Comment:
Here, i > 0 and output.timestamp > 0 seem like equivalent levels of comparison.
Wouldn't one conditional comparison be enough?

Reply:
Yes, it's redundant condition. I'll remove it.